src/Units/UnitContext.java

```diff
@@ -246,8 +250,16 @@ public int calculateVision()
       // if it's a surface unit, give it the boost the terrain would provide, so long as it's not adjacent-only vision
       if( model.isSurfaceUnit() )
         visionRange += env.terrainType.getVisionBoost();
-      if( env.weatherType == Weathers.RAIN )
+      if( env.weatherType == Weathers.RAIN && !CO.immuneToClouds )
         --visionRange;
+      if( env.weatherType == Weathers.SMOKE && !CO.immuneToClouds )
+      {
+        // SMOKE sets vision to 1 (before buffs but after mountains) in DoR, but Drake's version in DS only subtracts 1 from vision.
+        if( null != unit && unit.CO.gameRules.fogMode.dorMode )
+          visionRange = 1;
+        else // Default to Trilogy logic, since "has most of normal vision" will stick out more vs "vision = 1" than vice versa.
```

Comment:
I can't parse this as a coherent English sentence. You are saying the trilogy logic will stick out more, which typically connotes badness, but you also have it as the default, and then you have "a" vs "b" vs vice versa ("b" vs "a"?)

Reply:
"If badness occurs, the badness will be more obvious".
I like bugs being obvious-er

Reply:
Okay. I like bugs being obvious. Fail fast and all that. And I can infer from the code that you want to default to Trilogy behavior because you think it will be more noticeable when erroneously applied. I just don't get that from the comment.
"A sticks out more than B than vice versa" just reads like nonsense.

Reply:
"A will stick out more vs B than vice versa"
"A on the backdrop of B is more visible than the reverse"

Reply:
That makes sense. The structure of the comment sentence is just too tortured and terse to be sure that's what it means.

Reply:
I do enjoy terseness.

Reply:
And I commend clarity.